### ISSUE TITLE
FR: add a `clean` command

### ISSUE THREAD

**Issue by prbzrg** [JuliaLang/Pkg.jl#3672]
Having a command to delete `.julia` or part of it, like `compiled`, is very appreciating.
It would help in issues like https://github.com/SciML/LinearSolve.jl/issues/395

I know there is `rm(joinpath(homedir(), ".julia"))` but having a `clean` command help us document situations that removing part of `.julia` fix the problem.

**Comment by KristofferC** [comment]
I don't think this should ever be needed. If you need to clean out `.julia` to fix something it is a bug and the bug should be fixed. 

Also,

> having a clean command help us document situations that removing part of .julia fix the problem.

is quite unspecified. What parts of `.julia`?

**Comment by prbzrg** [comment]
Most of the problems are fixed now, we don't need this anymore.
